### PR TITLE
e2e cleanups

### PR DESCRIPTION
**Pull request by vrothberg**
A series of commits to speed up the e2e tests.  I focused on the pull tests but did not finish going through all of them due to time constraints. There is still much potential for speed ups and I bet there are more low-hanging fruits.

Please refer to the individual commits for change descriptions.

@containers/podman-maintainers PTAL

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by vrothberg** [comment]
@baude FYI

**Comment by rhatdan** [comment]
LGTM

**Comment by vrothberg** [comment]
Around 30 seconds faster on my machine. Since it's network bound and I am on fiber, I expect more noticeable speed ups in slower networks.

File: test/e2e/pull_test.go [hunk 1]
```diff
@@ -79,39 +84,36 @@ var _ = Describe("Podman pull", func() {
...
 		session.WaitWithDefaultTimeout()
 		Expect(session).Should(Exit(0))
 
-		// Without a tag/digest the input is normalized with the "latest" tag, see #11964
-		session = podmanTest.Podman([]string{"rmi", "testdigest_v2s2"})
+		session = podmanTest.Podman([]string{"images"})
 		session.WaitWithDefaultTimeout()
-		Expect(session).Should(Exit(1))
+		Expect(session).Should(Exit(0))
+		Expect(len(session.OutputToStringArray())).To(BeNumerically(">=", 2))
```
Comment (Luap99) [inline test/e2e/pull_test.go:110]:
```suggestion
		Expect(len(session.OutputToStringArray())).To(BeNumerically(">=", 2), "Expected at least two images")
```
nit, the error message would be something like expected number to be greater or equal than 2, yhis is not very useful for log readers. `To()` accepts an extra argument for a custom error description.

**Review by Luap99 — COMMENTED** [review]
LGTM, one nit

**Pull request merged**